Added Mocking For Tokens Endpoint in Tests

swierq opened the pull request:
aligning tests with DoLogin changes in: bef8ffb1bdc611e6c3785ad00ffa785a7589a386

Signed-off-by: Przemyslaw Swierczek <swierq@gmail.com>

raghavav commented on vrealize/resource.go:
```diff
@@ -259,8 +274,28 @@ func createResource(d *schema.ResourceData, meta interface{}) error {
+	}
+
+	//if both business group name and id are provided but does not belong to the same business group, throw an error
+	if len(business_group_name) > 0 && len(business_group_id) > 0 && businessGroupIdFromName != business_group_id {
```
To be picky about whitespace checks, consider using:

leng(strings.TrimSpace(...)) > 0 in all these checks

Prativa20 replied:
This might not work when the string has more than one word, say, the BG name is "Quality Engineering". Can do this check for id though.

raghavav replied:
No, TrimSpace only trims from the ends, not in the middle of strings.

Prativa20 replied:
Right!
Using TrimSpace while reading BG id and name.

raghavav commented on vrealize/resource.go:
```diff
@@ -930,3 +965,30 @@ func waitForRequestCompletion(d *schema.ResourceData, meta interface{}) error {
+// Retrieve business group id from business group name
+func (vRAClient *APIClient) GetBusinessGroupId(businessGroupName string) (string, error) {
+
+	path := "identity/api/tenants/" + vRAClient.Tenant + "/subtenants"
```
Have you checked if vRA already has an API to retrieve business group by name? It very well might. This should be commonly used functionality in the product.

Prativa20 replied:
Checked the vRA API doc, there is no such API for filtering based on name.

Prativa20 replied:
Using the filter-based OData query to get the BG object based on name

raghavav commented on vrealize/resource.go:
```diff
@@ -930,3 +965,23 @@ func waitForRequestCompletion(d *schema.ResourceData, meta interface{}) error {
+	apiError := new(APIError)
+	_, err := vRAClient.HTTPClient.New().Get(path).Receive(BusinessGroups, apiError)
+	if err != nil {
+		return "", err
```
better to return nil (and the error of course) in the event of an error, I think.

Prativa20 replied:
Here the return type is string(the BG id), which does not accept nil as a value.

raghavav replied:
plead my ignorance here. thanks.

swierq committed:
Added Mocking For Tokens Endpoint in Tests

aligning tests with DoLogin changes in: bef8ffb1bdc611e6c3785ad00ffa785a7589a386

Signed-off-by: Przemyslaw Swierczek <swierq@gmail.com>
